src/benchmarklib/tpch/tpch_query_generator.cpp

```diff
@@ -52,6 +70,339 @@ std::string TPCHQueryGenerator::get_preparation_queries() const {
+      parameters.emplace_back("'"s + type + "'");
+      break;
+    }
+
+    case 9 - 1: {
+      const auto color = colors.list[color_dist(random_engine)].text;
+
+      parameters.emplace_back("'%"s + color + "%'");
+      break;
+    }
+
+    case 10 - 1: {
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 23};
+      const auto diff = date_diff_dist(random_engine);
+      const auto begin_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff);
+      const auto end_date = calculate_date(boost::gregorian::date{1993, 01, 01}, (diff + 3));
+
+      parameters.emplace_back("'"s + begin_date + "'");
+      parameters.emplace_back("'"s + end_date + "'");
+      break;
+    }
+
+    case 11 - 1: {
+      const auto nation = nations.list[nation_dist(random_engine)].text;
+      const auto fraction = 0.0001 / (_scale_factor > 0 ? _scale_factor : 1);
+
+      parameters.emplace_back("'"s + nation + "'");
+      parameters.emplace_back(std::to_string(fraction));
+      parameters.emplace_back("'"s + nation + "'");
+      break;
+    }
+
+    case 12 - 1: {
+      const auto shipmode1 = l_smode_set.list[shipmode_dist(random_engine)].text;
+      std::string shipmode2;
+      do {
+        shipmode2 = l_smode_set.list[shipmode_dist(random_engine)].text;
+      } while (shipmode1 == shipmode2);
+
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 4};
+      const auto diff = date_diff_dist(random_engine);
+      const auto begin_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff * 12);
+      const auto end_date = calculate_date(boost::gregorian::date{1993, 01, 01}, (diff + 1) * 12);
+
+      parameters.emplace_back("'"s + shipmode1 + "'");
+      parameters.emplace_back("'"s + shipmode2 + "'");
+      parameters.emplace_back("'"s + begin_date + "'");
+      parameters.emplace_back("'"s + end_date + "'");
+      break;
+    }
+
+    case 13 - 1: {
+      const auto words1 = std::vector{"special", "pending", "unusual", "express"};
+      const auto words2 = std::vector{"packages", "requests", "accounts", "deposits"};
+
+      static thread_local std::uniform_int_distribution<> word_dist{0, 3};
+
+      parameters.emplace_back("'%"s + words1[word_dist(random_engine)] + '%' + words2[word_dist(random_engine)] + "%'");
+      break;
+    }
+
+    case 14 - 1: {
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 47};
+      const auto diff = date_diff_dist(random_engine);
+      const auto begin_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff);
+      const auto end_date = calculate_date(boost::gregorian::date{1993, 01, 01}, diff + 1);
+
+      parameters.emplace_back("'"s + begin_date + "'");
+      parameters.emplace_back("'"s + end_date + "'");
+      break;
+    }
+
+    case 15 - 1: {
+      auto query_15 = std::string{tpch_queries.at(15)};
+
+      static thread_local std::uniform_int_distribution<> date_diff_dist{0, 4};
```

Comment:
The standard says "DATE is the first day of a randomly selected month between the first month of 1993 and the 10th month of 1997.", but this seems to generate always to generate a first-of-January date?

Reply:
Oups. The PDF is currently not available. Same as in 4, isn't it?

Reply:
yep, same in 4